docs/src/proposals/scrambled-tx.md

```diff
@@ -0,0 +1,335 @@
+
+### Summary of Proposed Solution
+
+This proposal tries to combat against MEV, at the cost of (slight) global
+induced latency of cluster as a whole.
+
+At highest level, a full mitigation of MEV can be attained _if leaders (and
+searchers) can't know the contents of other's transactions until finishing to
+packing and ordering them into a new block_. In addition to this, the
+transaction details must also be deterministically revealed after the ordering
+of transactions is committed.
+
+So, naive 2-step transaction submission (encrypted tx, then its key) won't work,
+let alone increased latency.
+
+Instead, this proposal uses a special transaction construction _only to delay
+the revelation of its content for the short amount of time tied to confirmation
+by the cluster_. Specifically, transactions are encrypted by a private key
+derived from the recursive sha256, and its seed and cipher text are submitted to
+the cluster at the same time.
+
+This construction is called *scrambling* across this proposal.
+
+Conversely, the forced computation to reveal the transaction details by solving
+the recursive sha256 is called *descrambling*.
+
+### Summary of consensus changes
+
+In addition to the obvious change of submission of scrambled transactions by
+clients, this proposal introduces a new pipeline stage (called the descrambling
+stage) and splits the voting into 2 phases (pre-vote and vote).
+
+Firstly, the leader feeds incoming transactions down the turbine immediately
+with minimum sanitization (This assumes leaders are now so-called bank-less and
+scheduling is delayed and on-chain via mempool).
+
+The validators pre-vote on the completed block, once they recover all data.
+Thus, the opt-conf commitment of data availability of the block can be rather
+established rather quickly, without replay timing variation by all other
+validators (This average duration will be the basis of difficulty adjustment of
+scrambling).
+
+Also, rpc can notify optimistic execution signal to clients at this point. This
+signifies subsequent pipeline processing is fully deterministic and not reverted
+unless opt-conf is violated (some portion of stake is slash-able).
+
+At the same time, the validators also initiate the descrambling stage for the
+block in a random, collaborative and distributed manner. Once they solve their
+chosen chunk of scrambled transactions, they send revealed encryption keys to
+the leader to be propagated via turbine likewise the block shreds. Also
+sig-verify is done at the end of descrambling.
+
+After all transactions are descrambled from the cooperative efforts, the block
+can be finally scheduled and executed deterministically.
+
+Lastly, validators votes on the reproduced bank hash after replaying the
+processing likewise. Then, rpc can notify optimistic confirmation signal to
+clients.
+
+### Downsides
+
+- Increased latency
+- Some computational cost on clients
+- Reliance of local wall clock time
```

Comment:
Okay, i think we can lift this downside by exposing intra-block ledger entries' hashes to transaction.

basically, it should work more granular poh timestamp. so, transactions specify barely average single-trip-time to leader's worth number of future entries with referecened recent entryhash.